arvo/hoon.hoon

```diff
@@ -10079,7 +10061,16 @@
     ++  wisp                                            ::  core tail
       %-  ulva
       %+  cook
-        |=(a/(list {p/term q/foot}) (~(gas by *(map term foot)) a))
+        |=  a/(list {p/term q/foot})
+        =|  take/(set term)
+        =+  cur=a
+        |-
+          ?~  cur
+            (~(gas by *(map term foot) a))
```

Comment:
Or `(gas:by a b)` ;)

Is the `:door` (`|_`) an aesthetic thing (a la DRY), or is there some functional or performance advantage?